xdsl/dialects/onnx.py

```diff
@@ -26,6 +26,100 @@
 from xdsl.utils.exceptions import VerifyException
 
 
+def verify_unidirectional_broadcast_shape(
+    lhs: TensorType[Attribute], rhs: TensorType[Attribute], res: TensorType[Attribute]
+) -> None:
+    """
+    Returns a unidirectional broadcastable shape
+    """
+    lhs_shape = lhs.get_shape()
+    rhs_shape = rhs.get_shape()
+    expected_shape = unidirectional_broadcast_shape(list(lhs_shape), list(rhs_shape))
+    if not expected_shape:
+        raise VerifyException(
+            f"operands have incompatible shapes: {lhs_shape} and {rhs_shape}"
+        )
+    res_type_shape = list(res.get_shape())
```

Comment:
what happens when you make this change? I'm surprised you need `list` here and not above
```suggestion
    res_type_shape = res.get_shape()
```

Reply:
erm i believe since in the tests since its a list '[]`, it would fail, because res.get_shape() is a `tuple[int]' , thats what i believe it returns when i looked at the shaped class in the builtin

Reply:
correction: i mean when i looked at the tensortype class, that .get_shape() is a `tuple[int]` so i need the `list` so it matches

Reply:
Ah, I see, in that case I'd convert the list to a tuple in the check on the line below, but it's a nit at this point